### PR TITLE
Switch .NET tracer to injecting both base64 & binary headers

### PR DESCRIPTION
**Pull request by veerbia**
## Summary of changes
- Switch .NET tracer to injecting both base64 & binary headers
- Binary headers can be disabled via a config variable: DD_DATA_STREAMS_LEGACY_HEADERS=false (defaults to true)
- .NET tracer, when extracting the context from headers looks at both base64 & binary headers

## Reason for change
Data Streams previously used binary encoding in Kafka headers. This was causing issues in cross language communication because of the difference in negative byte handling. That’s why we switched to base64 encoding.
Today, .NET is the only remaining tracer using binary encoding for Kafka, SQS & RabbitMQ.

This is causing 3 issues:
- Communication between .NET & Java breaks, since Java doesn’t support negative bytes in Kafka headers
- Manual instrumentation breaks between .NET and any other language, since manual instrumentation uses base64 encoding
- .NET can’t consume payloads from any other tracers, since the other tracers encode in base64

Also, byte headers are causing a crash of the [.NET application](https://datadog.zendesk.com/agent/tickets/1824351) (not reproduced yet).

## Implementation details
- Injects base64-encoded headers by default to ensure cross-language compatibility.
- Binary headers are injected only if `DD_DATA_STREAMS_LEGACY_HEADERS` (default: true) is enabled for backward compatibility.
- Prefers base64 headers for extraction. If base64 is unavailable or malformed, it falls back to binary headers when legacy headers are enabled.
- A new configuration setting DD_DATA_STREAMS_LEGACY_HEADERS controls whether binary headers are included.

## Test coverage
I've added tests for the following cases:
- Inject_WhenLegacyHeadersDisabled_DoesNotIncludeBinaryHeader 
    - Ensures that when legacy headers are disabled, the binary header is not included in the injected headers.

- Extract_WhenBothHeadersPresent_PrefersBase64Header
    - Confirms that when both Base64 and binary headers are present, the Base64 header is preferred for extraction.

- InjectedHeaders_HaveCorrectFormat
    - Validates that injected headers are in the correct format, especially checking if the Base64-encoded header is valid and correctly decoded.

- InjectHeaders_WhenLegacyHeadersDisabled_DoesNotIncludeLegacyHeader 
    - Ensures that legacy headers are excluded when legacy header support is disabled.

- Inject_WhenLegacyHeadersEnabled_IncludesBothHeaders 
    - Confirms that both Base64 and binary headers are injected when legacy header support is enabled.

- Extract_WhenBase64HeaderIsMalformed_ReturnsFallbackToBinary 
    - Verifies that if the Base64 header is malformed, the system falls back to using the binary header for extraction.

### CODE REVIEW

File: tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/DataStreamsMonitoringKafkaTests.cs [hunk 1]
```diff
@@ -27,126 +28,72 @@ public DataStreamsMonitoringKafkaTests(ITestOutputHelper output)
...
-    ///    T1-->>-C1: Consume
-    ///    C1->>+T2a: Produce
-    ///    T2a-->>-C2a: Consume
-    ///    C2a->>+T3a: Produce
-    ///
-    ///    A->>+T2: Produce
-    ///    T2-->>-C2: Consume
-    ///    C2->>+T3: Produce
-    /// </summary>
-    /// <param name="enableConsumerScopeCreation">Is the scope created manually or using built-in support</param>
```
Comment (bouwkast) [inline tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/DataStreamsMonitoringKafkaTests.cs:59]: Was this no longer relevant? Wondering why it was removed

File: tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/DataStreamsMonitoringKafkaTests.cs
Comment (bouwkast) [inline tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/DataStreamsMonitoringKafkaTests.cs:1]: I'm a bit confused in this file, why are there so many changes? It seems that all of the old test logic was removed

I would've expected that `SetEnvironmentVariable(ConfigurationKeys.DataStreamsMonitoring.LegacyHeadersEnabled, enableLegacyHeaders ? "1" : "0");` is added along with `GetKafkaTestData()` and the previous tests would run one test where the legacy headers are enabled and one where they weren't.

File: tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/DataStreamsMonitoringRabbitMQTests.cs [hunk 2]
```diff
@@ -12,78 +12,89 @@
...
 using Xunit;
 using Xunit.Abstractions;
 
-namespace Datadog.Trace.ClrProfiler.IntegrationTests;
-
-[UsesVerify]
-[Trait("RequiresDockerDependency", "true")]
-public class DataStreamsMonitoringRabbitMQTests : TestHelper
+namespace Datadog.Trace.ClrProfiler.IntegrationTests
 {
```
Comment (bouwkast) [inline tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/DataStreamsMonitoringRabbitMQTests.cs:20]: Nitpicky

Changing from the top level namespace (no `{`) to the old way where there are `{` `}` makes the changes here in this file difficult to see.

**Review by bouwkast — APPROVED** [review]

File: tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/DataStreamsMonitoringKafkaTests.cs [hunk 3]
```diff
@@ -17,7 +19,6 @@
 namespace Datadog.Trace.ClrProfiler.IntegrationTests;
 
 [UsesVerify]
-[Collection(nameof(KafkaTests.KafkaTestsCollection))]
```
Comment (bouwkast) [inline tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/DataStreamsMonitoringKafkaTests.cs:20]: I don't think we want to remove this

**Comment by andrewlock** [comment]
## Throughput/Crank Report :zap:

Throughput results for AspNetCoreSimpleController comparing the following branches/commits:
- [This PR (6448)](https://github.com/DataDog/dd-trace-dotnet/tree/4c7e223356dcd81ad6105e3b698acfd9e5d3819a)
- [master](https://github.com/DataDog/dd-trace-dotnet/tree/7544783d76c0b8ed1a60e50a930648a5bcf611e5)
- [benchmarks/2.9.0](https://github.com/DataDog/dd-trace-dotnet/tree/89ae3f4ee7d85b824badda5873166707c9e560b4)

Cases where throughput results for the PR are worse than latest master (5% drop or greater), results are shown in **red**.

Note that these results are based on a _single_ point-in-time result for each branch. For full results, see [one](https://ddstaging.datadoghq.com/dashboard/fnz-afi-c2i/apm-net-crank) of the [many](https://ddstaging.datadoghq.com/dashboard/c5g-i7d-kcy/ciapp-apm-net-throughput-tests), [many](https://ddstaging.datadoghq.com/dashboard/uxh-m8j-qhi/ciapp-apm-net-throughput-tests-kevin) dashboards!

```mermaid
gantt
    title Throughput Linux x64 (Total requests) 
    dateFormat  X
    axisFormat %s
    section Baseline
    This PR (6448) (11.096M)   : 0, 11096382
    master (11.176M)   : 0, 11175779
    benchmarks/2.9.0 (11.033M)   : 0, 11032866

    section Automatic
    This PR (6448) (7.312M)   : 0, 7312245
    master (7.188M)   : 0, 7187812
    benchmarks/2.9.0 (7.786M)   : 0, 7785853

    section Trace stats
    master (7.592M)   : 0, 7592225

    section Manual
    master (11.202M)   : 0, 11201615

    section Manual + Automatic
    This PR (6448) (6.810M)   : 0, 6809676
    master (6.624M)   : 0, 6623622

    section DD_TRACE_ENABLED=0
    master (10.184M)   : 0, 10184081

```
```mermaid
gantt
    title Throughput Linux arm64 (Total requests) 
    dateFormat  X
    axisFormat %s
    section Baseline
    This PR (6448) (9.561M)   : 0, 9561017
    master (9.299M)   : 0, 9299402
    benchmarks/2.9.0 (9.495M)   : 0, 9494821

    section Automatic
    This PR (6448) (6.353M)   : 0, 6353019
    master (6.339M)   : 0, 6338804

    section Trace stats
    master (6.490M)   : 0, 6490416

    section Manual
    master (9.438M)   : 0, 9437615

    section Manual + Automatic
    This PR (6448) (5.940M)   : 0, 5940038
    master (5.979M)   : 0, 5979389

    section DD_TRACE_ENABLED=0
    master (8.785M)   : 0, 8784619

```
```mermaid
gantt
    title Throughput Windows x64 (Total requests) 
    dateFormat  X
    axisFormat %s
    section Baseline
    This PR (6448) (10.036M)   : 0, 10035673
    master (9.992M)   : 0, 9991636
    benchmarks/2.9.0 (10.020M)   : 0, 10019592

    section Automatic
    This PR (6448) (6.297M)   : 0, 6297190
    master (6.531M)   : 0, 6530893
    benchmarks/2.9.0 (7.255M)   : 0, 7255257

    section Trace stats
    master (7.153M)   : 0, 7152931

    section Manual
    master (10.055M)   : 0, 10055115

    section Manual + Automatic
    This PR (6448) (6.160M)   : 0, 6159823
    master (5.851M)   : 0, 5850726

    section DD_TRACE_ENABLED=0
    master (9.524M)   : 0, 9524080

```

**Comment by robertpi** [comment]
Replaces: https://github.com/DataDog/dd-trace-dotnet/pull/6257

**Review by bouwkast — APPROVED** [review]

**Review by lucaspimentel — APPROVED** [review]

**Pull request merged**